### ISSUE TITLE
Parent bone rotation ignored with transform noScale

### ISSUE THREAD

**Issue by esavichev** [pixijs/pixi-spine#127]
When using transform value "noscale" runtime should respect bone's parent rotation. The latest version does not. 

Bug reproduces on these animations for both old and new 3.5 format ([assets.zip](https://github.com/pixijs/pixi-spine/files/575104/assets.zip)) 2 directories are the exports of the same Spine project for runtimes 3.4.02 and 3.5.08-beta.

pixi.js - v4.1.1
Simple snippet to reproduce:

```javascript
window.addEventListener('load', function () {
    var renderer = PIXI.autoDetectRenderer(800, 600,{backgroundColor : 0x1099bb});
    document.body.appendChild(renderer.view);
    var stage = new PIXI.Container();

    PIXI.loader
            .add('shrimp34', 'assets/3.4/shrimp.json')
            .load(function (loader, resources) {
                createAnimations(resources.shrimp34.spineData, 100, 50, '3.4');
            });

    PIXI.loader
            .add('shrimp35', 'assets/3.5/shrimp.json')
            .load(function (loader, resources) {
                createAnimations(resources.shrimp35.spineData, 100, 250, '3.5');
            });

    function createAnimations(data, x, y, txt) {
        txt = stage.addChild(new PIXI.Text(txt)); txt.x = x; txt.y = y;

        for (var i = 0, len = data.animations.length; i < len; i++) {
            var mc = new PIXI.spine.Spine(data);
            stage.addChild(mc);
            mc.scale.set(0.03);
            mc.position.set(x + i * 100, y + 100);
            mc.state.setAnimation(0, data.animations[i].name, true);
        }
    }

    animate();
    function animate() {
        requestAnimationFrame(animate);
        renderer.render(stage);
    }
});

```

**Comment by ivanpopelyshev** [comment]
Seems like we have a problem. I'll take a loot at it in the evening. I guess its pixi-spine stuff, but if its in esotericSoftware's core, I'll ask them.

**Comment by esavichev** [comment]
Thanks for quick reply. There are 2 more issues with the same assets on 1.1.3 (I mean 3.4 export) older version.

1. If you take a look on "fright" animation, you will see an eye under the shrimp, while in Spine it is positioned where the eye should be. Looks like it ignores parent's (head) position and renders relative to root.

2. If you'll try to create Spine. 
Then run `mc.setAnimation(0, "idle", true);` 
Then later on the same instance (I did it from browser console) `mc.setAnimation(0, "fright", true);`
You'll see weird animations behavior. Try then to change animation back to "idle" an it gets even more weird. Maybe I misunderstood the usage of this method? Is this a bug or I use this method in wrong way? Sorry for asking here, but it's not clear for me from documentation.

Should I open new issues for these two?

**Comment by ivanpopelyshev** [comment]
If you dont want to wait for me, do the following:

Clone https://github.com/EsotericSoftware/spine-runtimes/tree/dev/spine-ts, make sure you checked out "dev" branch

Build it (optional, just to be sure)

```bash
tsc
tsc -p tsconfig-threejs.json
```

Modify threejs example that way it runs your animation. Also you can remove all rotations. instead of "raptor" everywhere should be your spine name.

If it works, try to set "mesh.skeleton.flipX=true" and see the same thing flipped. If it also works, then congratulations, that's pixi-spine bug.

If it behaves like pixi, then its core bug.

**Comment by esavichev** [comment]
Checked out dev branch of spine-ts and slightly modified webgl/example/test.html
Just changed assets and added click handler on document.body to switch animations via setAnimation method. Works fine at first glance. But better check it yourself, not sure if I've made no mistakes in branches etc...

**Comment by ivanpopelyshev** [comment]
Does it work if skeleton is flipped by Y? :)

**Comment by esavichev** [comment]
Oh, sorry, forgot to check... Yes, it works with flipX or flipY or even both of them.

Telling truth, I'm not quite sure that I've performed the tests on dev branch.
I used `svn checkout https://github.com/EsotericSoftware/spine-runtimes/branches/dev/spine-ts/` to checkout. My test works fine on 3.4 format. But when I test on 3.5 format it behaves just like pixi-spine 1.1.3 version: noScale is not working. It's expected behavior for runtime which does not understand "transform" property and renders just with "inheritScale" and "inheritRotation" as they default to true. So I'm a bit confused why does dev branch behaves like older runtime.

**Comment by ivanpopelyshev** [comment]
master is 3.4, dev is 3.5, lets work on 3.5 because 3.4 will be deprecated soon. Ok, i'll look at it in a few hours.

**Comment by ivanpopelyshev** [comment]
Please test it on https://dl.dropboxusercontent.com/u/46636530/pixi-spine/3.5/pixi-spine.js

**Comment by esavichev** [comment]
Unfortunately, no. Dev branch of spine-ts works as expected with 3.5 format.
Latest pixi-spine still ignores parent bone rotation. And something is wrong with setAnimation.

Please, take a look at attached sources to compare the expected result and current on pixi-spine. I used latest dev version of spine-webgl.js and your latest version of pixi-spine. 
[pixi_spine_issue127.zip](https://github.com/pixijs/pixi-spine/files/577754/pixi_spine_issue127.zip)


In spine-ts-demo.html animation works right. Click it to toggle via setAnimation - works as expected.

In index,html child bones have wrong rotation. Hover any animation to trigger setAnimation on it. Big white eyes used in "fright" animation appear and stay forever even in animations where they should be invisible.

**Comment by ivanpopelyshev** [comment]
Fixed in 1.3.1

**Comment by esavichev** [comment]
Yes, now rotations are correct. This bug is fixed. But there are still 2 issues which are reproducible in the same sources I've attached before. I'll describe them by opening new issues, as they don't relate to this issue topic.
